### PR TITLE
Move to GraphQL

### LINKED ISSUE #150: ⭐️ Verisure deprecated REST API: Error at startup
**Summary of the problem;** Verisure have changed the hostnames to their API.

---

I noticed that i have trouble with getting the verisure child bridge up and running. it loops the error message you can see down below. i tried to start it without child bridge, but that just looped homebridge in to a restart.
it worked fine yesterday.

```bash
[15/03/2023, 09:55:35] [homebridge-verisure] Restarting Process...
[15/03/2023, 09:55:35] [Ebeco] Sensor config: floor
[15/03/2023, 09:55:36] [homebridge-verisure] Launched child bridge with PID 17299
[15/03/2023, 09:55:36] Registering platform 'homebridge-verisure.verisure'
[15/03/2023, 09:55:36] [homebridge-verisure] Loaded homebridge-verisure v1.16.0 child bridge successfully
[15/03/2023, 09:55:36] Loaded 0 cached accessories from cachedAccessories.0E96B5C2DDC6.

/var/lib/homebridge/node_modules/homebridge-verisure/node_modules/axios/lib/core/createError.js:16
  var error = new Error(message);
              ^
Error: Request failed with status code 404
    at createError (/var/lib/homebridge/node_modules/homebridge-verisure/node_modules/axios/lib/core/createError.js:16:15)
    at settle (/var/lib/homebridge/node_modules/homebridge-verisure/node_modules/axios/lib/core/settle.js:17:12)
    at IncomingMessage.handleStreamEnd (/var/lib/homebridge/node_modules/homebridge-verisure/node_modules/axios/lib/adapters/http.js:269:11)
    at IncomingMessage.emit (node:events:525:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
[15/03/2023, 09:55:36] [homebridge-verisure] Child bridge process ended
[15/03/2023, 09:55:36] [homebridge-verisure] Process Ended. Code: 1, Signal: null
```

### PR DESCRIPTION
**Pull request by ptz0n**
Thanks again for all your support, this branch is now ready for testing.

**Update the plugin globally**

```bash
$ npm install -g homebridge-verisure@2.0.0-rc.4
```

**… or run it from a working dir**

```bash
$ git clone https://github.com/ptz0n/homebridge-verisure.git
$ cd homebridge-verisure
$ git checkout graphql
$ npm install
$ npm start
```

**For configuration, see updated [README.md](https://github.com/ptz0n/homebridge-verisure/blob/graphql/README.md).** Until released, please use:

```bash
$ ./path/to/updated/homebridge-verisure/bin/verisure
```

---

- [x] Platform
- [x] Alarm accessory
- [x] Climate sensor accessory
- [x] Contact sensor accessory
- [x] Door lock accessory
  - [ ] Auto lock switch (not working properly)
  - [ ] Volume switch (not working properly)
- [x] Smart plug accessory

Fixes #150

### CODE REVIEW

**Comment by larsxsamuelsson** [comment]
Running into MFA errors after reconfiguring with NPX.
Only log message is [2023-03-19 15:02:40] [Verisure] MFA is enabled for user. Please see README.

And then Verisure sends me another MFA code.
If i remove the password part of the config alltogether, it fails with a 401 instead.

**Comment by larsxsamuelsson** [comment]
> Running into MFA errors after reconfiguring with NPX. Only log message is [2023-03-19 15:02:40] [Verisure] MFA is enabled for user. Please see README.
> 
> And then Verisure sends me another MFA code. If i remove the password part of the config alltogether, it fails with a 401 instead.

Uhm
This might actually be on me. Running hb-service and I think its playing tricks on the manual update of the plugin.

**Comment by larsxsamuelsson** [comment]
Deleted my earlier comments as it was my own fault.
After setting things up properly, everything works as expected. Awesome job.

**Comment by ptz0n** [comment]
Great @larsxsamuelsson and thanks for taking the time to try it out. Is there any step of the setup that should be corrected or could be clarified?

**Comment by priisagersoegaard** [comment]
Tested with Yale Doorman v2 and lock/unlock works but the autolock on/off function have status “no response” in HomeKit.

**Comment by larsxsamuelsson** [comment]
@ptz0n  Not really. As soon as you push it it will be upgraded the "normal way" for people using Homebridge UI and hb-service I guess. So it shouldn't provide any problems. Except for them having to do a new npx-setup.

**Comment by ptz0n** [comment]
@priisagersoegaard Interesting, anything more descriptive in the logs? I'll have a second look to see if there is something off with any of those queries.

**Comment by priisagersoegaard** [comment]
@ptz0n 
No I can't se anything...

Only info: [19.3.2023 19.18.13] [verisure] Østergade SmartLock (Frontdoor): Getting current auto lock config.

**Comment by helangen** [comment]
I have everything working except the two V2N locks. I get "Accessory not responding" in Homekit.

**Comment by priisagersoegaard** [comment]
@helangen I Think you have the same problem as me since I can see you also have auto lock enabled. 
It’s the auto lock function which not are responding.

**Comment by ptz0n** [comment]
Version `2.0.0-rc.1` is now available, please update and share your logs when changing auto lock and volume switches.

**Comment by helangen** [comment]
> @helangen I Think you have the same problem as me since I can see you also have auto lock enabled. It’s the auto lock function which not are responding.

Yes, I figured that out after I posted. Updating to RC1 now.

**Comment by helangen** [comment]
> Version `2.0.0-rc.1` is now available and include debug logging. Please update, set log level to `debug` and share your logs when changing auto lock and volume switches.

I still have "Not responding". The switches don't react.

[3/19/2023, 10:02:37 PM] [homebridge-verisure] Hjemme Kjellerdør: Getting current sensor state.
[3/19/2023, 10:02:38 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Got config: undefined
[3/19/2023, 10:02:38 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got config: undefined
[3/19/2023, 10:02:55 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Getting current audio config.
[3/19/2023, 10:02:56 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Getting current auto lock config.
[3/19/2023, 10:02:56 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got config: undefined

**Comment by torandreroland** [comment]
I can confirm that I am able to set the on characteristic successfully for the auto-lock and audio level switch services (I have to do it in the app Controller for HomeKit as it's not possible do change a switch in app Apple Home when the service is not responding).

However getting the characteristic is still not functioning as of v2.0.0-rc.1.

**Comment by ptz0n** [comment]
There we have something...

```
Hjemme SmartLock (Kjellerdør): Got config: undefined
```

**Comment by ptz0n** [comment]
@torandreroland Great! Then I believe we have pinned it down to getting the current configuration, which is used to resolve the state of these switches.

**Might be a crazy idea;** should we release version `2.0.0` without the auto lock and volume switches?

**Comment by helangen** [comment]
> @torandreroland Great! Then I believe we have pinned it down to getting the current configuration, which is used to resolve the state of these switches.
> 
> **Might be a crazy idea;** should we release version `2.0.0` without the auto lock and volume switches?

You could, but it does work without for the most part. I would have no issues with it.

File: lib/accessories/doorlock.js [hunk 1]
```diff
@@ -143,10 +143,10 @@ class DoorLock extends VerisureAccessory {
   getAudioState(callback) {
     this.log('Getting current audio config.');
 
-    this.installation.client({ url: this.configUrl })
```
Comment (ptz0n) [inline lib/accessories/doorlock.js:129]: Pretty silly. Should work better passing in an operation.

**Comment by ptz0n** [comment]
Might have found the root issue; please update to version `2.0.0-rc.2`.

**Comment by helangen** [comment]
> I can confirm that I am able to set the on characteristic successfully for the auto-lock and audio level switch services (I have to do it in the app Controller for HomeKit as it's not possible do change a switch in app Apple Home when the service is not responding). I have checked in the Verisure web-interface that the updated characteristic is reflected on Verisure's end.
> 
> However getting the characteristic is still not functioning as of v2.0.0-rc.1.

Using the same app and changing the state I get this in the log:

[3/19/2023, 10:25:05 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Setting auto lock to: true

**Comment by helangen** [comment]
> Might have found the root issue; please update to version `2.0.0-rc.2`.

On it.

**Comment by ptz0n** [comment]
Silly me. Hold on…

```
Unable to get installations. Please check configured credentials: Request failed with status code 401
```

**Comment by helangen** [comment]
> Silly me. Hold on…
> 
> ```
> Unable to get installations. Please check configured credentials: Request failed with status code 401
> ```

Same here

**Comment by ptz0n** [comment]
Version `2.0.0-rc.3` ready for testing.

**Comment by helangen** [comment]
> Version `2.0.0-rc.3` ready for testing.

Installing...

**Comment by helangen** [comment]
> > Version `2.0.0-rc.3` ready for testing.
> 
> Installing...

Update: Still no go. 

[3/19/2023, 10:43:39 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Got config: [{"device":{"area":"Hoveddør","deviceLabel":"2ANH WMET","__typename":"Device"},"configuration":{"autoLockEnabled":true,"voiceLevel":"NORMAL","volume":"LOW","__typename":"YaleLockConfiguration"},"__typename":"SmartLock"}]
[3/19/2023, 10:43:39 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Got config: [{"device":{"area":"Hoveddør","deviceLabel":"2ANH WMET","__typename":"Device"},"configuration":{"autoLockEnabled":true,"voiceLevel":"NORMAL","volume":"LOW","__typename":"YaleLockConfiguration"},"__typename":"SmartLock"}]
[3/19/2023, 10:43:39 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got config: [{"device":{"area":"Kjellerdør","deviceLabel":"2CU9 26YP","__typename":"Device"},"configuration":{"autoLockEnabled":true,"voiceLevel":"NORMAL","volume":"LOW","__typename":"YaleLockConfiguration"},"__typename":"SmartLock"}]
[3/19/2023, 10:43:39 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got config: [{"device":{"area":"Kjellerdør","deviceLabel":"2CU9 26YP","__typename":"Device"},"configuration":{"autoLockEnabled":true,"voiceLevel":"NORMAL","volume":"LOW","__typename":"YaleLockConfiguration"},"__typename":"SmartLock"}]
[3/19/2023, 10:43:50 PM] [homebridge-verisure] Hjemme Alarm (Hjemme): Getting current alarm state.
[3/19/2023, 10:43:50 PM] [homebridge-verisure] Hjemme Alarm (Hjemme): Getting current alarm state.
[3/19/2023, 10:43:51 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Getting current auto lock config.
[3/19/2023, 10:43:51 PM] [homebridge-verisure] Hjemme Hoveddør: Getting current sensor state.
[3/19/2023, 10:43:51 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Got config: [{"device":{"area":"Hoveddør","deviceLabel":"2ANH WMET","__typename":"Device"},"configuration":{"autoLockEnabled":true,"voiceLevel":"NORMAL","volume":"LOW","__typename":"YaleLockConfiguration"},"__typename":"SmartLock"}]
[3/19/2023, 10:43:59 PM] [homebridge-verisure] Hjemme Alarm (Hjemme): Getting current alarm state.
[3/19/2023, 10:44:00 PM] [homebridge-verisure] Hjemme Hoveddør: Getting current sensor state.
[3/19/2023, 10:44:03 PM] [homebridge-verisure] Hjemme Kjellerdør: Getting current sensor state.
[3/19/2023, 10:44:06 PM] [homebridge-verisure] Hjemme Kjøkkendør: Getting current sensor state.
[3/19/2023, 10:44:07 PM] [homebridge-verisure] Hjemme VoiceBox (Hall): Getting current temperature value.
[3/19/2023, 10:44:09 PM] [homebridge-verisure] Hjemme Siren (Kjøkken): Getting current temperature value.
[3/19/2023, 10:44:09 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Getting current audio config.
[3/19/2023, 10:44:09 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got config: [{"device":{"area":"Kjellerdør","deviceLabel":"2CU9 26YP","__typename":"Device"},"configuration":{"autoLockEnabled":true,"voiceLevel":"NORMAL","volume":"LOW","__typename":"YaleLockConfiguration"},"__typename":"SmartLock"}]
[3/19/2023, 10:44:09 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Getting current auto lock config.
[3/19/2023, 10:44:10 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Got config: [{"device":{"area":"Hoveddør","deviceLabel":"2ANH WMET","__typename":"Device"},"configuration":{"autoLockEnabled":true,"voiceLevel":"NORMAL","volume":"LOW","__typename":"YaleLockConfiguration"},"__typename":"SmartLock"}]
[3/19/2023, 10:44:10 PM] [homebridge-verisure] Hjemme Smoke detector (Master bedroom): Getting current temperature value.
[3/19/2023, 10:44:10 PM] [homebridge-verisure] Hjemme Smoke detector (Hall 2 etg): Getting current temperature value.
[3/19/2023, 10:44:11 PM] [homebridge-verisure] Hjemme Smoke detector (Stue): Getting current temperature value.
[3/19/2023, 10:44:11 PM] [homebridge-verisure] Hjemme Smoke detector (Kjeller): Getting current temperature value.
[3/19/2023, 10:44:11 PM] [homebridge-verisure] Hjemme Smoke detector (Hall): Getting current temperature value.
[3/19/2023, 10:44:13 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Setting auto lock to: true

**Comment by ptz0n** [comment]
I believe we got it! Version `v2.0.0-rc.4`.

**Comment by helangen** [comment]
> I believe we got it! Version `v2.0.0-rc.4`.

Installing now. Will update you again.

**Comment by helangen** [comment]
> > I believe we got it! Version `v2.0.0-rc.4`.
> 
> Installing now. Will update you again.

UPDATE:

YAY!

Now the "not responding" is gone. Seems fine. Will continue to test.

[3/19/2023, 11:02:47 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Getting current auto lock config.
[3/19/2023, 11:02:47 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Getting target lock state.
[3/19/2023, 11:02:47 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Getting current lock state.
[3/19/2023, 11:02:47 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Getting current audio config.
[3/19/2023, 11:02:48 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got config: {"device":{"area":"Kjellerdør","deviceLabel":"2CU9 26YP","__typename":"Device"},"configuration":{"autoLockEnabled":true,"voiceLevel":"NORMAL","volume":"SILENCE","__typename":"YaleLockConfiguration"},"__typename":"SmartLock"}
[3/19/2023, 11:02:48 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got config: {"device":{"area":"Kjellerdør","deviceLabel":"2CU9 26YP","__typename":"Device"},"configuration":{"autoLockEnabled":true,"voiceLevel":"NORMAL","volume":"SILENCE","__typename":"YaleLockConfiguration"},"__typename":"SmartLock"}
[3/19/2023, 11:02:59 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Setting auto lock to: true
[3/19/2023, 11:03:01 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Setting auto lock to: false

**Comment by helangen** [comment]
> > > I believe we got it! Version `v2.0.0-rc.4`.
> > 
> > 
> > Installing now. Will update you again.
> 
> UPDATE:
> 
> YAY!
> 
> Now the "not responding" is gone. Seems fine. Will continue to test.
> 
> [3/19/2023, 11:02:47 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Getting current auto lock config. [3/19/2023, 11:02:47 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Getting target lock state. [3/19/2023, 11:02:47 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Getting current lock state. [3/19/2023, 11:02:47 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Getting current audio config. [3/19/2023, 11:02:48 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got config: {"device":{"area":"Kjellerdør","deviceLabel":"2CU9 26YP","__typename":"Device"},"configuration":{"autoLockEnabled":true,"voiceLevel":"NORMAL","volume":"SILENCE","__typename":"YaleLockConfiguration"},"__typename":"SmartLock"} [3/19/2023, 11:02:48 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got config: {"device":{"area":"Kjellerdør","deviceLabel":"2CU9 26YP","__typename":"Device"},"configuration":{"autoLockEnabled":true,"voiceLevel":"NORMAL","volume":"SILENCE","__typename":"YaleLockConfiguration"},"__typename":"SmartLock"} [3/19/2023, 11:02:59 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Setting auto lock to: true [3/19/2023, 11:03:01 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Setting auto lock to: false

UPDATE2:

The volume function does not seem to do anything. Switching does not change anything in the logs.

**Comment by torandreroland** [comment]
I can confirm that both get and set of both auto-lock and audio level is reflected on Verisure's end. I think it's safe to publish a final v2.0 version.

Good work, @ptz0n!!!

**Comment by helangen** [comment]
I can confirm the same now. Don't know why it did not work before. Still nothing in the log when changing the audio.

**Comment by ptz0n** [comment]
Time to 🚢

**Comment by helangen** [comment]
Confirm logging of audio for Yale is now working for v2.0.0.

[3/20/2023, 11:14:01 AM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Setting audio volume to: SILENCE
[3/20/2023, 11:14:03 AM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Setting audio volume to: LOW

**Comment by ptz0n** [comment]
Thank you @helangen and @torandreroland for your efforts in testing and debugging with me. We did it!

**Comment by helangen** [comment]
btw. I find the Homekit integration less responsive than before. I get "not responding" a lot of times and it seems the app is the same. Does everyone else have the same or is it my setup that is the problem? I was wondering if it was due to the new changes by Verisure.

**Comment by ptz0n** [comment]
@helangen Agree, something is off. I'll start debugging and might raise a new issue to discuss this specifically.

**Comment by larsxsamuelsson** [comment]
@ptz0n @helangen fwiw it’s not happening here. Don’t have that lock you were on about though.

Did you guys upgrade to the new homekit architecture debacle of iOS 16.2? Because after I did get that upgrade through, I’ve had more or less no  ”not responding” issues.

**Comment by helangen** [comment]
> @ptz0n @helangen fwiw it’s not happening here. Don’t have that lock you were on about though.
> 
> Did you guys upgrade to the new homekit architecture debacle of iOS 16.2? Because after I did get that upgrade through, I’ve had more or less no ”not responding” issues.

Check your sensors as well. They update slow. I am running iOS 16.3.1.

**Comment by helangen** [comment]
I restarted the child-bridge with verisure and they are responding again. I locked the door now and saw the following in the logs:

[3/20/2023, 9:10:34 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Setting target lock state to: 1
[3/20/2023, 9:10:34 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState
[3/20/2023, 9:10:34 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState
[3/20/2023, 9:10:35 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState
[3/20/2023, 9:10:35 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState
[3/20/2023, 9:10:35 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState
[3/20/2023, 9:10:35 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState
[3/20/2023, 9:10:35 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState
[3/20/2023, 9:10:35 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState
[3/20/2023, 9:10:36 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState
[3/20/2023, 9:10:36 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState
[3/20/2023, 9:10:36 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState
[3/20/2023, 9:10:36 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState
[3/20/2023, 9:10:36 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState
[3/20/2023, 9:10:37 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState
[3/20/2023, 9:10:37 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState
[3/20/2023, 9:10:37 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState
[3/20/2023, 9:10:37 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState
[3/20/2023, 9:10:37 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "OK" back from: pollLockState

**Comment by helangen** [comment]
> I restarted the child-bridge with verisure and they are responding again. I locked the door now and saw the following in the logs:
> 
> [3/20/2023, 9:10:34 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Setting target lock state to: 1 [3/20/2023, 9:10:34 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState [3/20/2023, 9:10:34 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState [3/20/2023, 9:10:35 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState [3/20/2023, 9:10:35 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState [3/20/2023, 9:10:35 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState [3/20/2023, 9:10:35 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState [3/20/2023, 9:10:35 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState [3/20/2023, 9:10:35 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState [3/20/2023, 9:10:36 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState [3/20/2023, 9:10:36 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState [3/20/2023, 9:10:36 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState [3/20/2023, 9:10:36 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState [3/20/2023, 9:10:36 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState [3/20/2023, 9:10:37 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState [3/20/2023, 9:10:37 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState [3/20/2023, 9:10:37 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "null" back from: pollLockState [3/20/2023, 9:10:37 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Resolving: pollLockState [3/20/2023, 9:10:37 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Got "OK" back from: pollLockState

It locks fine though, but something is off. It seems that it stops responding after a while. Anyone else with similar problems?

**Comment by ptz0n** [comment]
@helangen Interesting. The locks seem quite slow to receive the command and communicate back once done.

We might want to set the state directly when getting a response from the command (Setting target lock state to: 1), before starting to poll.

If the lock is jammed, that state will reflect in Home after a few seconds as a new state.

**Comment by ptz0n** [comment]
@helangen Which value will you see from pollLockState in the logs if you try to lock a door that's not quite closed? I'm guessing a value similar to `jammed` instead of `OK`.

**Comment by helangen** [comment]
> @helangen Which value will you see from pollLockState in the logs if you try to lock a door that's not quite closed? I'm guessing a value similar to `jammed` instead of `OK`.

I will check.

**Comment by helangen** [comment]
> This is what I got after trying to lock it in a jammed state a few times. Then I closed it properly and locked. Seems like 'null' is what I get. I have changed the poll interval to 120 btw. Trying to increase battery life.
> 
> [3/20/2023, 10:36:19 PM] [homebridge-verisure] This plugin slows down Homebridge. The read handler for the characteristic 'Lock Current State' was slow to respond! See https://homebridge.io/w/JtMGR for more info. [3/20/2023, 10:36:19 PM] [homebridge-verisure] This plugin slows down Homebridge. The read handler for the characteristic 'Lock Target State' was slow to respond! See https://homebridge.io/w/JtMGR for more info. [3/20/2023, 10:36:19 PM] [homebridge-verisure] This plugin slows down Homebridge. The read handler for the characteristic 'On' was slow to respond! See https://homebridge.io/w/JtMGR for more info. [3/20/2023, 10:36:19 PM] [homebridge-verisure] This plugin slows down Homebridge. The read handler for the characteristic 'On' was slow to respond! See https://homebridge.io/w/JtMGR for more info. [3/20/2023, 10:36:19 PM] [homebridge-verisure] This plugin slows down Homebridge. The read handler for the characteristic 'Lock Current State' was slow to respond! See https://homebridge.io/w/JtMGR for more info. [3/20/2023, 10:36:19 PM] [homebridge-verisure] This plugin slows down Homebridge. The read handler for the characteristic 'Lock Target State' was slow to respond! See https://homebridge.io/w/JtMGR for more info. [3/20/2023, 10:36:19 PM] [homebridge-verisure] This plugin slows down Homebridge. The read handler for the characteristic 'On' was slow to respond! See https://homebridge.io/w/JtMGR for more info. [3/20/2023, 10:36:19 PM] [homebridge-verisure] This plugin slows down Homebridge. The read handler for the characteristic 'On' was slow to respond! See https://homebridge.io/w/JtMGR for more info. [3/20/2023, 10:36:20 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Got "null" back from: pollLockState [3/20/2023, 10:36:21 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Resolving: pollLockState [3/20/2023, 10:36:21 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Got "null" back from: pollLockState [3/20/2023, 10:36:21 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Setting target lock state to: 1 [3/20/2023, 10:36:21 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Resolving: pollLockState [3/20/2023, 10:36:22 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Resolving: pollLockState [3/20/2023, 10:36:22 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Got "null" back from: pollLockState [3/20/2023, 10:36:23 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Resolving: pollLockState [3/20/2023, 10:36:23 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Getting current audio config. [3/20/2023, 10:36:23 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Getting target lock state. [3/20/2023, 10:36:23 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Getting current auto lock config. [3/20/2023, 10:36:23 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Getting current audio config. [3/20/2023, 10:36:23 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Getting target lock state. [3/20/2023, 10:36:23 PM] [homebridge-verisure] Hjemme SmartLock (Kjellerdør): Getting current auto lock config. [3/20/2023, 10:36:23 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Got "null" back from: pollLockState [3/20/2023, 10:36:23 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Resolving: pollLockState [3/20/2023, 10:36:23 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Got "OK" back from: pollLockState [3/20/2023, 10:36:25 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Got "null" back from: pollLockState [3/20/2023, 10:36:25 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Resolving: pollLockState [3/20/2023, 10:36:25 PM] [homebridge-verisure] Hjemme SmartLock (Hoveddør): Got "OK" back from: pollLockState

**Comment by priisagersoegaard** [comment]
> 

I have the same issue. 
After a while my lock also get in stage “no response”

**Comment by torandreroland** [comment]
Same here.

**Comment by helangen** [comment]
Yeah, @ptz0n said he raise a separate issue

**Pull request merged**